itaki/rudi
# Keyboard device enhancements

Keyboard devices can now create dynamic events that are mapped to individual keys, as defined in their config preferences

Testings this:
- Stash or commit whatever you have going locally
- Pull down the `keyboard-device-enhancements` branch
- Replace your config with `keyboard-super-simple-led.json`
- See that keys `q` and `w` start and stop the fake mitre saw
- See that keys `a` and `s` start and stop the fake dust collector
- Change the config to use a letter twice and see that there's a logged warning and that the second mapping is ignored


More info:
After this PR is approved and you merge `main` into the `simple-pi-button` branch the `keyboard-shop-light.json` config should work (but can't really be tested until merged)
